CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift

```diff
@@ -171,23 +171,15 @@ internal class CRFHeartRateSampleProcessor {
             // Need to keep 2 extra seconds due to filtering lopping off the first 2 seconds of data.
             let meanOrder = self.meanFilterOrder(for: roundedRate)
             let windowLength = Int(CRFHeartRateWindowSeconds + 2) * roundedRate + meanOrder
-            if self.pixelSamples.count >= windowLength {
+            if self.pixelSamples.count >= windowLength, self.isValidSamplingRate(roundedRate) {
```

Comment:
@syoung-smallwisdom Do we have the ability to impute missing values on the fly here given the continuous nature of the signal so that we can get to a constant sampling rate?

Reply:
@DanWebster seems like a bit of a chicken-and-egg thing--we won't have a good idea what to predict for the missing raw values until we've analyzed the raw data. I suppose if it's just an occasional missing sample, we could maybe linearly interpolate between the two adjacent ones; if it's more egregious than that though, it gets trickier.